doc/content/design/trusted-certificates.md

```diff
@@ -0,0 +1,151 @@
+This case is outside the scope of trusted certificates managed by XAPI and is included here only for completeness.
+* An XAPI process on one host initiates a TLS connection to an XAPI process on another host within the same pool.
+This case is covered in the base design and is listed here for completeness.
+* An XAPI process initiates a TLS connection to an external service, such as an appliance.
+This case benefits from the improvements introduced in this design.
+* A non-XAPI process (like licensing agent) running on a host managed by XAPI initiates a TLS connection to an external service, such as a License Server.
+This case benefits from the improvements introduced in this design as well.
+
+
+# Changes
+## Database schema
+The *Certificate* class in database is defined to represent general certificates, including trusted certificates.
+One existing class field "type" supports the following enumeration values:
+* "ca": trusted certificates including both root CA and peer.
+* "host": identity certificate of a host for communication with entities outside the pool.
+* "host_internal": identity certificate of a host for communication with other pool members.
+
+Two improvements in this design:
+* A new value "peer" is introduced in this design so that the existing "ca" now represents trusted root CA only.
+The new "peer" will represent trusted peer certificates.
+
+* A new enumeration type "purpose" is introduced to indicate the intended usage of a trusted certificate.
+A new *Certificate* class field "purpose" (a set of values of enumeration type "purpose") will be added to represent all applicable purposes of a trusted certificate.
+By default, this set is empty which corresponds to the existing "ca" certificates for general purpose.
+
+## API
+
+### pool.install_ca_certificate
+
+This is an existing API to install a trusted certificate into the pool with its arguments being defined as:
+* session (ref session_id): reference to a valid session;
+* name (string): the name of the certificate;
+* cert (string): the certificate in PEM format.
+
+Prior to this design, the API's name parameter represents the certificate file name as persisted on the dom0 file system.
+In this design, this API will be deprecated because it exposes implementation details that should remain internal and hidden from users.
+The new "pool.install_trusted_certificate" should be used instead.
+For the same reason, "pool.uninstall_ca_certificate" will also be deprecated.
+
+### pool.install_trusted_certificate
+This is a new API introduced in this design with its arguments being defined as:
+* session (ref session_id): reference to a valid session;
+* self (ref Pool): reference to the pool;
+* ca (boolean): the trusted certificate is a root CA certificate used to verify a chain (true), or a peer certificate used for certificate pinning (false);
+* cert (string): the trusted certificate in PEM format;
+* purpose (string list): the purposes of the trusted certificate.
```

Comment:
Is it possible to upload multiple trusted certificates for a single purpose? 
I am asking as Domain Controller may have different certificates. 

And I think it is possible as we handle filename in the filesystem internally.